src/main/java/data/schedule/DayList.java

```diff
@@ -0,0 +1,265 @@
+     *
+     * @param planForDay Plan name scheduled for a particular day.
+     * @return planForDayWith Plan name scheduled for a particular day with added empty spaces at the front and back.
+     */
+    public String padWithSpaces(String planForDay) {
+        String prePadded = "";
+        String postPadded = "";
+        int numberOfSpacesToBePadFront = (FIXED_LENGTH - planForDay.length()) / 2;
+        int numberOfSpacesToBePadBack = FIXED_LENGTH - prePadded.length() - planForDay.length();
+        for (int i = 0; i < numberOfSpacesToBePadFront; i++) {
+            prePadded += " ";
+        }
+        for (int i = 0; i < numberOfSpacesToBePadBack; i++) {
+            postPadded += " ";
+        }
+
+        String planForDayWithPadding = prePadded + planForDay + postPadded;
+        return planForDayWithPadding;
+    }
+
+    /**
+     * Clear the plan for a specific day. The plan scheduled for the day will be cleared/reset.
+     * When a plan is clear/reset, the day will set to be a rest day.
+     *
+     * @param userArgument Day number to be reset/cleared.
+     * @throws InvalidScheduleException If the day number specified in userArgument is invalid.
+     */
+    public String clearDayPlan(String userArgument) throws InvalidScheduleException {
+        int dayNumber = Integer.parseInt(userArgument);
+        String className = this.getClass().getSimpleName();
+        if (!isDayValid(dayNumber)) {
+            logger.log(Level.WARNING, "User entered an invalid day number.");
+            throw new InvalidScheduleException(className, InvalidScheduleException.DAY_NUMBER_OUT_OF_RANGE);
+        }
+        if (dayList[dayNumber - 1] == null) {
+            return " ";
+        }
+
+        clearPlan(dayNumber - 1);
+        String dayName = covertDayNumberToDay(dayNumber);
+        return dayName;
+    }
+
+    /**
+     * Clear all the plans schdeule for the week. Reset the schedule list to default.
+     * All plans for the week will be set as rest day.
+     */
+    public void clearAllSchedule() {
+        for (int i = 0; i < NUMBER_OF_SCHEDULE_DAYS; i += 1) {
+            clearPlan(i);
+            assert (scheduleList[i] == null);
+            assert (dayList[i] == null);
+        }
+    }
+
+    /**
+     * Remove plan on the stated day.
+     */
+    public void clearPlan(int dayNumber) {
+        scheduleList[dayNumber] = null;
+        dayList[dayNumber] = null;
+    }
+
+    /**
+     * Covert the day number to its english meaning. For example,
+     * day number 1 will be converted to Monday,
+     * day number 2 will be converted to Tuesday,
+     * and so on till day number 7 which will be converted to Sunday.
+     *
+     * @param dayNumber Day number to be converted to its String equivalent.
+     * @return day English meaning of the day number specified.
+     */
+    public String covertDayNumberToDay(int dayNumber) {
```

Comment:
same code appears in Day class toString. so... why not reuse in one of the places?